src/mame/heathkit/h89.cpp

```diff
@@ -23,85 +47,195 @@
 
 #include "cpu/z80/z80.h"
 #include "machine/ins8250.h"
+#include "machine/ram.h"
 #include "machine/timer.h"
 
 namespace {
 
 
-#define H89_CLOCK (XTAL(12'288'000) / 6)
-#define INS8250_CLOCK (XTAL(1'843'200))
-
 class h89_state : public driver_device
 {
 public:
-	h89_state(const machine_config &mconfig, device_type type, const char *tag) :
-		driver_device(mconfig, type, tag),
-		m_maincpu(*this, "maincpu"),
-		m_tlb(*this, "tlb"),
-		m_console(*this, "console")
+	h89_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_maincpu_region(*this, "maincpu")
+		, m_mem_view(*this, "rom_bank")
+		, m_ram(*this, RAM_TAG)
+		, m_floppy_ram(*this, "floppyram")
+		, m_tlb(*this, "tlb")
+		, m_console(*this, "console")
+		, m_serial1(*this, "serial1")
+		, m_serial2(*this, "serial2")
+		, m_serial3(*this, "serial3")
+		, m_gpp(0)
+		, m_rom_enabled(true)
+		, m_timer_intr_enabled(true)
+		, m_floppy_ram_wp(true)
 	{
 	}
 
 	void h89(machine_config &config);
 
 private:
 	required_device<cpu_device> m_maincpu;
+	required_memory_region m_maincpu_region;
+	memory_view m_mem_view;
+	required_device<ram_device> m_ram;
+	required_device<ram_device> m_floppy_ram;
```

Comment:
Don’t use a `ram_device` for the floppy RAM, use a `required_shared_ptr<uint8_t>`.  The only purpose of the RAM device is for main RAM that can have the size set from command line/INI file.